segment_coverage/segment_coverage/main.py

```diff
@@ -133,22 +109,32 @@ def create_coverage_map(self, quality, all_hour_holes, all_hour_partials,
 		all_hour_holes -- a dict mapping hours to lists of holes
 		all_hour_holes -- a dict mapping hours to lists of partial segments
 		pixel_length -- length of a pixel in seconds
-		hour_count -- number of hours to create the map for
 		rows -- the height of the image"""
 
-		self.logger.info('Creating coverage map for {}'.format(quality))
 
-		latest_hour = datetime.datetime.strptime(max(all_hour_holes.keys()), HOUR_FMT)
-		hours = [latest_hour - datetime.timedelta(hours=i) for i in range(hour_count - 1, -1, -1)]
+		if self.first_hour is None:
+			first_hour = datetime.datetime.strptime(min(all_hour_holes.keys()), HOUR_FMT)
+		else:
+			first_hour = self.first_hour.replace(minute=0, second=0, microsecond=0)
+		if self.last_hour is None:
+			last_hour = datetime.datetime.strptime(max(all_hour_holes.keys()), HOUR_FMT)
+		else:
+			last_hour = self.last_hour.replace(minute=0, second=0, microsecond=0)
+		self.logger.info('Creating coverage map for {} from {} to {}'.format(quality,
+			first_hour.strftime(HOUR_FMT), last_hour.strftime(HOUR_FMT)))
+
+		hours = [first_hour]
+		latest_hour = first_hour
+		while latest_hour < last_hour:
```

Comment:
This should be <=, or else hours won't include last_hour

Reply:
It does since the value of latest_hour that is appended to the hours list is one hour plus the value of latest_hour that is compared in the while statement.

Reply:
ah, right. my mistake. maybe add comment explaining?